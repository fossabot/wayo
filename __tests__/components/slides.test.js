import 'babel-polyfill';
import { shallowMount,createLocalVue } from '@vue/test-utils';
import { render } from '@vue/server-test-utils';
import Slides from '../../src/components/slides';
import Indicator from '../../src/components/slides/_indicator.vue';

const LocalVue = createLocalVue();

describe('Logic', () => {
  const Wrapper = shallowMount(Slides, {
    localVue: LocalVue,
    propsData: {
      autoplay: false,
      loop: false
    }
  });
  test('computed', () => {
    expect(Wrapper.vm.isLoop).toBe(false);

    Wrapper.setProps({
      autoplay: true
    });
    expect(Wrapper.vm.isLoop).toBe(true);
    
    Wrapper.setProps({
      autoplay: false,
      loop: true
    });
    expect(Wrapper.vm.isLoop).toBe(true);

    Wrapper.setProps({
      autoplay: false,
      loop: true,
      animation: 'none'
    });
    Wrapper.setData({
      panStartX: 100,
      panCurrentX: 50
    });
    expect(Wrapper.vm.manualTranslate).toBe(50);

    Wrapper.setProps({
      autoplay: false,
      loop: false
    });
    Wrapper.setData({
      panStartX: 100,
      panCurrentX: 50,
      activeIndex: 2,
      itemCount: 3
    });
    expect(Wrapper.vm.manualTranslate).toBe(0);

    Wrapper.setData({
      panStartX: 20,
      panCurrentX: 50,
      activeIndex: 0
    });
    expect(Wrapper.vm.manualTranslate).toBe(0);
  });
});

describe('Snapshot', () => {
  test('slides', () => {
    const Wrapper = render(Slides, {
      localVue: LocalVue
    });
    expect(Wrapper).toMatchSnapshot();
  });

  test('slides item', () => {
    const Wrapper = render(Slides.Slide, {
      localVue: LocalVue
    });
    expect(Wrapper).toMatchSnapshot();
  });

  test('slides indicator', () => {
    const Wrapper = render(Indicator, {
      localVue: LocalVue,
      propsData: {
        index: 0
      }
    });
    expect(Wrapper).toMatchSnapshot();
  });
});