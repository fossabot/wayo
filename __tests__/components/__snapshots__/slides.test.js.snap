// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot slides 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "wayo-slides",
      "data-server-rendered": "true",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Snapshot slides indicator 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "wayo-slides__indicator wayo-slides__indicator_type_circle",
      "data-server-rendered": "true",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Snapshot slides item 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "wayo-slide__contentitem",
      "data-server-rendered": "true",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
